packages/server/src/arpa_reporter/lib/audit-report.js

```diff
@@ -154,24 +158,73 @@ async function generate(requestHost) {
     XLSX.utils.book_append_sheet(workbook, sheet2, 'Project Summaries');
 
     return {
+        periodId,
         filename: `audit report ${moment().format('yy-MM-DD')}.xlsx`,
         outputWorkBook: XLSX.write(workbook, { bookType: 'xlsx', type: 'buffer' }),
     };
 }
 
+function getS3Client() {
+    /*
+        TODO: Move this client generation to a separate module that handles creating all AWS clients.
+        Captured in ticket here: https://github.com/usdigitalresponse/usdr-gost/issues/1161
+    */
+
+    let s3;
+    if (process.env.LOCALSTACK_HOSTNAME) {
```

Comment:
Is it possible to configure the S3 client to point at localstack without this conditional, purely with environment variables? It seems a bit safer to have the uploader be totally unaware of the difference between localstack S3 and real S3 and have the configuration be purely driven via the environment, eg. test or dev and `.env`.

Reply:
Unfortunately (and frustratingly), [no, at least not natively (yet)](https://github.com/aws/aws-sdk/issues/229). However, a more general pattern could be to create a centralized AWS SDK config object to reuse when instantiating any service client. It wouldn't obviate the need for the conditional check, but it would provide consistent behavior divorced from any specific use-case. We could also choose a more generic environment variable name like `$AWS_ENDPOINT_URL` to make it less specific to LocalStack and something that wouldn't need to be wrapped in a conditional. But no AWS SDK for any language seems to provide an option to configure this automatically.

This seems like something worth exploring in (or as a follow-up to) #653 (cc @jakekreider).